### ISSUE TITLE
Different sensitivity analysis results depending on group of parameters being varied

### ISSUE THREAD

**Issue by abdullahhamadeh** [Open-Systems-Pharmacology/MoBi#475]
Here the sensitivity of Q to nu and to nu_part_hyd changes according to whether or not the two parameters are varied with another parameter of the model

[sensitivity-analysis-bug.zip](https://github.com/Open-Systems-Pharmacology/MoBi/files/4612719/sensitivity-analysis-bug.zip)

**Comment by StephanSchaller** [comment]
Yes, this may well be.

e.g. the sensitivity of fat organ volume may increase if the drug is more lipophilic.

**Comment by abdullahhamadeh** [comment]
Thanks.  I understand that that scenario can emerge as a second order effect in a global sensitivity analysis.  But in mobi the sensitivity analysis is only first order, isn't it? ie each parameter is perturbed from its nominal value separately from the others, and this process is done parameter by parameter.  In such a case I would expect that the sensitivity of the output to a parameter to be the same even if multiple parameters have been selected for perturbation in the analysis.

**Comment by StephanSchaller** [comment]
@Yuri05, you added the "bug" label. Have you already checked?

**Comment by Yuri05** [comment]
>But in mobi the sensitivity analysis is only first order, isn't it? ie each parameter is perturbed from its nominal value separately from the others, and this process is done parameter by parameter. In such a case I would expect that the sensitivity of the output to a parameter to be the same even if multiple parameters have been selected for perturbation in the analysis.

Yes. Seems to be a bug.

**Comment by msevestre** [comment]
@Yuri05 Any idea as to what is happening? Does SimModel keep some memory of previous values?

**Comment by Yuri05** [comment]
I don't think so. Interestingly the wrong sensitivity is completely zero.
![grafik](https://user-images.githubusercontent.com/25061876/81734872-79ef0680-9494-11ea-8625-0349056c8b09.png)

**Comment by msevestre** [comment]
Looking at this now...

**Comment by msevestre** [comment]
So nu is not only a function, but also a function of  nu_part_hyd

So when we vary nu, the function of nu_part_hyd is overwritten. 
Therefor the sensitivity of nu_part_hyd is then 0 as it is not used in the model anymore

@Yuri05 I am not sure what we should do here. Varying formula parameter does not make sense with the current implementation as we are defacto removing the formula

This is also one of the reason why the Add All constant does not add those parameters by default

**Comment by StephanSchaller** [comment]
Could you group the parameters for the SA and first do the constants and then the formulas, but here again, formulas can depend on formulas and you may get the same result..

But would the formulas not be reset to their original formula after they have been varied?

**Comment by msevestre** [comment]
@ their original formula after they have been varied?
No. We create a population if you want where all parameters are equal to the parameter before the variation except for one entry

Say you vary P1, P2, P3 with original values 1,2,3 and we have 10% variation for each parameter

The "Population" used for SA will look as follow

| P1 | P2 | P3 |
| -- |  -- | -- |
|0.9| 2 | 3|
|1.1|2|3|
|1|1.8|3|
|1|2.2|3|
|1|2|2.4|
|1|2|3.6|

e.g each parameter will be overwritten. This is ok for constant parameters but creates an issue when varying formula parameters.

Maybe we should rethink this but this is not mega easy

**Comment by StephanSchaller** [comment]
okay, got it, yes, sth to ponder ;-)

**Comment by Yuri05** [comment]
@msevestre 
>I am not sure what we should do here. Varying formula parameter does not make sense with the current implementation as we are defacto removing the formula

True. In v10 we could show an exclamation mark for all **formula** parameters selected
<img width="450" src="https://user-images.githubusercontent.com/25061876/81913850-d22b2300-95d0-11ea-90e3-fc2b2db7308a.png">

Trying to calculate all sensitivities correctly **in case some formula parameters were selected** would require in general to estimate the hierarchy level of all selected parameters and perform sensitivity calculations for each level independently.
